autotest/utilities/test_gdalwarp_lib.py

```diff
@@ -3171,7 +3171,7 @@ def test_gdalwarp_lib_automatic_grid_sampling():
         outputBounds=[-7655830, -6385994, 7152182, 8423302],
         dstSRS="+proj=laea +lat_0=48.514 +lon_0=-145.204 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs",
     )
-    assert ds.GetRasterBand(1).Checksum() == 46790
+    assert ds.GetRasterBand(1).Checksum() == 35573
```

Comment:
In addition to the checksum we should check the resolution returned by ds.GetGeoTransform()

Reply:
About this test, the resolution is very different with this patch but I checked the result visually and it looks ok to me.

GT without this patch (rounded to int):
(-7655830, 30850, 0.0, 8423302, 0.0, -30852)

GT with this patch (rounded to int):
(-7655830, 19433, 0.0, 8423302, 0.0, -19434)

do you think it is acceptable?

Reply:
> do you think it is acceptable?

looks reasonable